### DISCUSSION TITLE
Make Automatic `chown` can be disabled.

### DISCUSSION THREAD

**Discussion by lemniskett** [serversideup/docker-php#269] [Ideas & Feature Requests]
# 👉 Describe the problem
[comment]: <> (What are you trying to solve?)
When using an application that stores a lot of files in the filesystem (e.g. Nextcloud), the `chown` process in `src/cli/etc/s6-overlay/scripts/runas-user` will slow down the startup.

# 🏆 How to solve this problem
[comment]: <> (Describe the feature that you are proposing and how it will solve the problem)

### Add environment variable that skips the `chown` in `src/cli/etc/s6-overlay/scripts/runas-user`
Perhaps a variable name like "`WEBUSER_HOME_SKIP_CHOWN`"?
```
if [[ -n "$WEBUSER_HOME_SKIP_CHOWN" ]]; then
    # Make sure the home directory is updated with the proper UID & GID
    chown webuser:webgroup $WEBUSER_HOME
fi
```

# 🥰 Describe the "impact" on users?
[comment]: <> (How will this make people's lives better once it is solved?)
Users with a lot of files in the filesystem can disable the chown process for faster startup time.

**Comment by jaydrogers** [comment]
Hey @lemniskett,

Try using the beta images and let me know your thoughts. I don't have the `runas-user` script in there anymore and I don't use `chown` on start up: https://github.com/serversideup/docker-php/discussions/254

**Reply by lemniskett** [reply]
Hi @jaydrogers,

It seems like PUID and PGID variable is not supported in the beta images? I tried setting those variables but the output of `system("id")` always like this:

```
uid=33(www-data) gid=33(www-data) groups=33(www-data) uid=33(www-data) gid=33(www-data) groups=33(www-data)
```

**Reply by jaydrogers** [reply]
Instead of changing the ID on start up, it's best if you change the ID on build. See this for more information: https://github.com/serversideup/docker-php/issues/179#issuecomment-1871385674
